xcube/webapi/viewer/context.py

```diff
@@ -2,18 +2,50 @@
 # Permissions are hereby granted under the terms of the MIT License:
 # https://opensource.org/licenses/MIT.
 
+from contextlib import contextmanager
 from functools import cached_property
+from pathlib import Path
 from typing import Optional
 from collections.abc import Mapping
+import sys
 
+from chartlets import Extension
+from chartlets import ExtensionContext
 import fsspec
 
+from xcube.constants import LOG
+from xcube.server.api import Context
 from xcube.webapi.common.context import ResourcesContext
+from xcube.webapi.viewer.contrib import Panel
+
+Extension.add_contrib_point("panels", Panel)
 
 
 class ViewerContext(ResourcesContext):
     """Context for xcube Viewer API."""
 
+    def __init__(self, server_ctx: Context):
+        super().__init__(server_ctx)
+        self.ext_ctx: ExtensionContext | None = None
+
+    def on_update(self, prev_context: Optional[Context]):
+        super().on_update(prev_context)
+        viewer_config: dict = self.config.get("Viewer")
+        if viewer_config:
+            augmentation: dict | None = viewer_config.get("Augmentation")
+            if augmentation:
+                extension_refs: list[str] = augmentation["Extensions"]
+                path: Path | None = None
+                if "Path" in augmentation:
+                    path = Path(augmentation["Path"])
+                    if not path.is_absolute():
+                        path = Path(self.base_dir) / path
+                with prepend_sys_path(path):
```

Comment:
I feel as though there ought to be a more elegant way to do this without changing the system path globally, but on closer examination it seems that alternative approaches would need quite a lot of import system surgery :(.

Reply:
If you know anything better, please report back to me!